py_gtfs_rt_ingestion/lib/convert_gtfs_rt.py

```diff
@@ -197,40 +255,50 @@ def drill_entity(drill_keys: str) -> Optional[dict]:
 
         return record
 
-    def gz_to_pyarrow(self, filename: str) -> Tuple[datetime, pyarrow.table]:
+    def gz_to_pyarrow(
+        self, filename: str
+    ) -> Tuple[Optional[datetime], str, Optional[pyarrow.table]]:
         """
         Accepts filename as string. Converts gzipped json -> pyarrow table.
         Will handle Local or S3 filenames.
         """
-        filename = filename.replace("s3://", "")
-        with self.active_fs.open_input_stream(filename) as file:
-            json_data = json.load(file)
-
-        # Create empty 'table' as dict of lists for export schema
-        table = self.detail.empty_table()
-
-        # parse timestamp info out of the header
-        feed_timestamp = json_data["header"]["timestamp"]
-        timestamp = datetime.fromtimestamp(feed_timestamp, timezone.utc)
-
-        # for each entity in the list, create a record, add it to the table
-        for entity in json_data["entity"]:
-            record = self.record_from_entity(entity=entity)
-            record.update(
-                {
-                    "year": timestamp.year,
-                    "month": timestamp.month,
-                    "day": timestamp.day,
-                    "hour": timestamp.hour,
-                    "feed_timestamp": feed_timestamp,
-                }
-            )
+        try:
+            file_system = current_thread().__dict__["file_system"]
+            filename = filename.replace("s3://", "")
```

Comment:
slight modification in this function to use file system that was initialized for thread